lib/rubocop/cop/style/optional_arguments.rb

```diff
@@ -17,14 +20,30 @@ module Style
       #
       #   def foobar(a = 1, b = 2, c = 3)
       #   end
+      #
+      # @example Max: 3 (default)
+      #   # good
+      #   def foo(a = 1, b = 2, c = 3)
+      #   end
+      #
+      # @example Max: 2
+      #   # bad
+      #   def foo(a = 1, b = 2, c = 3)
+      #   end
+      #
       class OptionalArguments < Base
+        include ConfigurableMax
+
         MSG = 'Optional arguments should appear at the end ' \
               'of the argument list.'
+        TOO_MANY_MSG = 'Method has too many optional arguments. [%<count>d/%<max>d]'
```

Comment:
It should say "parameters". In general you can fix the terminology everywhere except in the cop's name, as we're checking the definition not the call sites.

Reply:
Yeah, that is correct that those are "parameters"  and not "arguments". But renaming it to "parameters", I think, may introduce some confusion, because this cop name includes "arguments" in its name, `def on_def(node)` - `node` also has `#arguments` method (and looks like `parser` gem does not have a notion of "parameters", but "arguments" everywhere - for calls and definitions). And seems like `rubocop`s own codebase uses "arguments" in most of the places where really it is about "parameters". Personally, I'm using both words interchangeably, and the real meaning depends on the context.
So should I fix the terminology?

Reply:
Those are fair points, although I'd disagree that RuboCop uses mostly arguments (just going over the names of the cops shows otherwise). Not that big of a deal though, it's just I never use the terms interchangeably and this terminology confuses me. I guess I'm one old grumpy coder. :D